python/pyarrow/tests/test_dataset.py

```diff
@@ -2258,11 +2254,6 @@ def test_write_dataset_use_threads(tempdir):
         use_threads=False
     )
 
-    # check that all files are the same in both cases
-    paths1 = [p.relative_to(target1) for p in target1.rglob("*")]
-    paths2 = [p.relative_to(target2) for p in target2.rglob("*")]
-    assert set(paths1) == set(paths2)
```

Comment:
Why was this removed? (does it no longer hold?)

Reply:
It no longer holds consistently; the auto incremented `{i}` doesn't necessarily round trip